packages/manager/apps/web/client/app/domain/zone/activate/activate.controller.js

```diff
@@ -1,43 +1,125 @@
-          value: this.domainName,
-        },
-      ],
-    };
-
-    this.$window.open(
-      `${this.constants.ORDER_URL[this.user.ovhSubsidiary]}${JSURL.stringify([
-        order,
-      ])}`,
-      '_blank',
+    // Auto-select duration
+    this.prices = filter(
+      this.serviceOption.prices,
+      (price) => price.pricingMode === 'default',
+    );
+    [this.price] = this.prices;
+    this.interval = this.price.interval;
+    this.isEditable = true;
+    if (this.prices.length === 1) {
+      // Go directly to the next step
+      this.currentIndex = 1;
+      this.isEditable = false;
+    }
+  }
+
+  resetCart() {
+    if (this.cart) {
+      this.cart = undefined;
+      this.cartId = undefined;
+    }
+  }
+
+  prepareCheckout() {
+    if (!this.cart && !this.checkoutLoading) {
+      this.checkoutLoading = true;
+      this.prepareOrderCart(this.price)
+        .then((cart) => {
+          this.cart = cart;
+          this.cartId = cart.cartId;
+        })
+        .catch((error) =>
+          this.Alerter.error(
+            this.$translate.instant('web_domain_zone_activation_error', {
+              message: get(error, 'data.message', error),
+            }),
+            'app.domain.product.zoneactivate',
+          ),
+        )
+        .finally(() => (this.checkoutLoading = false));
+    }
+  }
+
+  checkout() {
+    this.checkoutOrderCart(
+      this.autoPayWithPreferredPaymentMethod,
+      this.cartId,
+      this.price.value === 0,
     );
-    return this.goBack(
-      this.$translate.instant('web_domain_zone_activation_success'),
+  }
+
+  checkoutOrderCart(autoPayWithPreferredPaymentMethod, cartId, isOptionFree) {
+    this.checkoutLoading = true;
+    this.DomainDnsZoneActivateService.checkoutOrderCart(
+      autoPayWithPreferredPaymentMethod,
+      cartId,
+    )
+      .then((order) => {
+        if (isOptionFree) {
+          this.goBack(
+            this.$translate.instant('web_domain_zone_activation_order_success'),
+          );
+        } else {
+          this.$window.open(order.url, '_blank');
+          this.goBack(
+            this.$translate.instant('web_domain_zone_activation_success', {
+              t0: order.url,
+            }),
+          );
+        }
+      })
+      .catch((error) =>
+        this.Alerter.error(
+          this.$translate.instant('web_domain_zone_activation_error', {
+            message: get(error, 'data.message', error),
+          }),
+          'app.domain.product.zoneactivate',
+        ),
+      )
+      .finally(() => {
+        this.checkoutLoading = false;
+      });
+  }
+
+  prepareOrderCart(price) {
+    return this.DomainDnsZoneActivateService.prepareOrderCart(
+      this.user.ovhSubsidiary,
+    ).then((cart) =>
+      this.DomainDnsZoneActivateService.addItemToCart(
+        cart.cartId,
+        this.serviceName,
+        this.serviceOption,
+        price,
+      ).then((finalCart) => {
+        finalCart.cartId = cart.cartId;
```

Comment:
```suggestion
  return { ...finalCart, cartId: cart.cartId }
```
to avoid updating the parameter